text_generation/causal_lm/cpp/continuous_batching/library/src/continuous_batching_pipeline.cpp

```diff
@@ -232,11 +248,14 @@ class ContinuousBatchingPipeline::Impl {
         std::vector<GenerationResult> results;
         results.reserve(m_requests.size());
 
-        while (has_running_requests()) {
+        while (has_running_requests() && !out_of_memory()) {
             std::vector<GenerationResult> partial_results = step();
-            results.insert(results.end(), partial_results.begin(), partial_results.end());
+            if (partial_results.size() > 0)
+                results.insert(results.end(), partial_results.begin(), partial_results.end());
         }
 
+        OPENVINO_ASSERT(!out_of_memory(), "Not enough memory for processing the requests.");
```

Comment:
instead of aborting server execution, let's just safely return information that we cannot handle some requests.

Reply:
Done.